backend/app/models/uni_models.py

```diff
@@ -226,35 +223,90 @@ class Config:
         json_encoders = {ObjectId: str}
         schema_extra = {
             "example": {
-                "name": "AGU",
+                "name": "Test University",
+                "website": "test.com",
+                "country": "USA",
+                "city": "Los Angeles",
+                "address": "123 Main Street",
+                "phone": "555-555-5555",
+                "email": "info@test.com",
+                "zip_code": "90210",
+                "description": "Test University Description",
+                "logo": "test_logo.png",
+                "cover_photo": "test_cover_photo.png",
             }
         }
 
 
 class UniversityAPIModel(BaseModel):
     id: Optional[str] = Field(alias="_id")
     name: Optional[str]
-    curSemesterID: Optional[str]
+    website: Optional[str]
+    country: Optional[str]
+    city: Optional[str]
+    address: Optional[str]
+    phone: Optional[str]
+    email: Optional[str]
+    zip_code: Optional[str]
+    description: Optional[str]
+    logo: Optional[str]
+    cover_photo: Optional[str]
+    current_semester_id: Optional[str]
 
     class Config:
         allow_population_by_field_name = True
-        arbitrary_types_allowed = True
-        json_encoders = {ObjectId: str}
         schema_extra = {
             "example": {
                 "_id": "61fc266ae3d749b1d65c17c6",
-                "name": "AGU",
-                "curSemesterID": "61fc266ae3d749b1d65c17c7",
+                "name": "Test University",
+                "website": "test.com",
+                "country": "USA",
+                "city": "Los Angeles",
+                "address": "123 Main Street",
+                "phone": "555-555-5555",
+                "email": "info@test.com",
+                "zip_code": "90210",
+                "description": "Test University Description",
+                "logo": "test_logo.png",
+                "cover_photo": "test_cover_photo.png",
+                "current_semester_id": "61fc266ae3d749b1d65c17c7",
             }
         }
 
 
-class UpdateUniversityNameModel(BaseModel):
-    name: str = Field(...)
+class UpdateUniversityModel(BaseModel):
```

Comment:
If we want them to be optional, we should find a way for MongoDB to update few of them or we can iterate through the attributes of the model and if it is not none we can update it. Which one will be better?

Reply:
If we want them to be optional, we should find a way for MongoDB to update few of them or we can iterate through the attributes of the model and if it is not none we can update it. Which one will be better?

Reply:
> class UpdateUniversityModel(BaseModel):
>     name: Optional[str] = Field(min_length=2, max_length=100)
>     website: Optional[str] = Field(min_length=5, max_length=255)
>     country: Optional[str] = Field(min_length=2, max_length=100)
>     city: Optional[str] = Field(min_length=2, max_length=100)
>     address: Optional[str] = Field(min_length=2, max_length=100)
>     phone: Optional[str] = Field(min_length=2, max_length=100)
>     email: Optional[str] = Field(min_length=2, max_length=100)
>     zip_code: Optional[str] = Field(min_length=2, max_length=100)
>     description: Optional[str] = Field(max_length=255)
>     logo: Optional[str] = Field(max_length=255)
>     cover_photo: Optional[str] = Field(max_length=255)
> 
>     class Config:
>         schema_extra = {
>             "example": {
>                 "name": "Test University",
>                 "website": "test.com",
>                 "country": "USA",
>                 "city": "Los Angeles",
>                 "address": "123 Main Street",
>                 "phone": "555-555-5555",
>                 "email": "info@test.com",
>                 "zip_code": "90210",
>                 "description": "Test University Description",
>                 "logo": "test_logo.png",
>                 "cover_photo": "test_cover_photo.png",
>             }
>         }

# IN UPDATE ROUTER

> updated_features = {}
> for key in university:
>        if university[key] is not None:
>             updated_features.update({key: university[key]})
> 
> update_result = await request.app.mongodb["universities"].update_one(
>        {"_id": unid}, {"$set": updated_features}
> )

This can be a solution to partially updating models. In the update_XXX models we can make everything Optional and then we can update not None fields in the update router. I can add this feature to other update endpoints as well.